dspace-uclouvain/src/main/java/org/dspace/uclouvain/consumer/CataretroFacultyMetadataConsumer.java

```diff
@@ -0,0 +1,159 @@
+package org.dspace.uclouvain.consumer;
+
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.*;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.event.Consumer;
+import org.dspace.event.Event;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.core.model.Entity;
+import org.dspace.uclouvain.core.model.EntityType;
+import org.dspace.uclouvain.factories.UCLouvainServiceFactory;
+import org.dspace.uclouvain.services.UCLouvainEntityService;
+import org.dspace.workflow.WorkflowItemService;
+import org.dspace.workflow.factory.WorkflowServiceFactory;
+import org.dspace.xmlworkflow.storedcomponents.XmlWorkflowItem;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Consumer to generate additional metadata from the faculty code ONLY for "catareto" collection.
+ *
+ * Authored-By: Renaud Michotte <renaud.michotte@uclouvain.be>
+ */
+public class CataretroFacultyMetadataConsumer implements Consumer {
+
+    private static Logger log = org.apache.logging.log4j.LogManager.getLogger(CataretroFacultyMetadataConsumer.class);
+
+    private String facultyCodeFieldname;
+    private String facultyNameFieldname;
+
+    private ItemService itemService;
+    private WorkspaceItemService workspaceItemService;
+    private WorkflowItemService<?> workflowItemService;
+    private CollectionService collectionService;
+    private MetadataFieldService metadataFieldService;
+    private UCLouvainEntityService uclouvainEntityService;
+
+    @Override
+    public void initialize() throws Exception {
```

Comment:
'throws Exception' not necessary since nothing can throw an error in this method.